arbutil/hash.go

```diff
@@ -2,15 +2,32 @@ package arbutil
 
 import (
 	"github.com/ethereum/go-ethereum/common"
-	"golang.org/x/crypto/sha3"
+	"github.com/ethereum/go-ethereum/crypto"
 )
 
-// StorageSlotAddress pads each argument to 32 bytes, concatenates and returns
-// keccak256 hashe of the result.
-func StorageSlotAddress(args ...[]byte) []byte {
-	hash := sha3.NewLegacyKeccak256()
+// PaddedKeccak256 pads each argument to 32 bytes, concatenates and returns
+// keccak256 hash of the result.
+func PaddedKeccak256(args ...[]byte) []byte {
+	var data []byte
 	for _, arg := range args {
-		hash.Write(common.BytesToHash(arg).Bytes())
+		data = append(data, common.BytesToHash(arg).Bytes()...)
 	}
-	return hash.Sum(nil)
+	return crypto.Keccak256(data)
+}
+
+// SumBytes sums two byte slices and returns the result.
+// If the sum of bytes are over 32 bytes, it return last 32.
+func SumBytes(a, b []byte) []byte {
+	// Normalize lengths to hash length.
+	a = common.BytesToHash(a).Bytes()
+	b = common.BytesToHash(b).Bytes()
+
+	sum := make([]byte, common.HashLength)
+	c := 0
+	for i := common.HashLength - 1; i >= 0; i-- {
```

Comment:
Here instead of the loop we can use big Int as well right ?

Something like this 

       bigA = bigA.SetBytes(a)
	bigB = bigB.SetBytes(b)
	bigC := bigA.Add(bigA, bigB)

	return common.BytesToHash(bigC.Bytes()).Bytes()

Dont have any strong preference either way.

Reply:
Good point, changed to that!